pandas/tests/frame/methods/test_astype.py

```diff
@@ -567,6 +569,11 @@ def test_astype_empty_dtype_dict(self):
         "df",
         [
             DataFrame(Series(["x", "y", "z"], dtype="string")),
+            DataFrame(Series(["x", "y", "z"], dtype="string[python]")),
+            pytest.param(
+                DataFrame(Series(["x", "y", "z"], dtype="string[pyarrow]")),
+                marks=td.skip_if_no("pyarrow", min_version="1.0.0"),
+            ),
```

Comment:
This didn't work while breaking some changes off #40679, in so in #40749 instead. so these changes won't be part of this PR.

(it seems pytest tries to create the DataFrame, before skipping, resulting in an ImportError... and we want to preserve that behavior https://github.com/pandas-dev/pandas/pull/39908#discussion_r585581525

cc @jreback

Reply:
> (it seems pytest tries to create the DataFrame, before skipping, resulting in an ImportError.

I think that is expected, since the DataFrame is a function parameter value, which is evaluated when creating the param object

Reply:
yeah, obvious when you stop and think about it.